tmva/sofie/src/RModel_GNN.cxx

```diff
@@ -0,0 +1,235 @@
+      nodes_update_block = std::move(other.nodes_update_block);
+      globals_update_block = std::move(other.globals_update_block);
+
+      edge_node_agg_block = std::move(other.edge_node_agg_block);
+      edge_global_agg_block = std::move(other.edge_global_agg_block);
+      node_global_agg_block = std::move(other.node_global_agg_block);
+
+      num_nodes = std::move(other.num_nodes);
+      num_edges = std::move(other.num_edges);
+      senders = std::move(other.senders);
+      receivers = std::move(other.receivers);
+
+      fName = std::move(other.fName);
+      fFileName = std::move(other.fFileName);
+      fParseTime = std::move(other.fParseTime);
+
+      return *this;
+    }
+
+    RModel_GNN::RModel_GNN(const GNN_Init& graph_input_struct){
+        edges_update_block.reset((graph_input_struct.edges_update_block).get());
+        nodes_update_block.reset((graph_input_struct.nodes_update_block).get());
+        globals_update_block.reset((graph_input_struct.globals_update_block).get());
+
+        edge_node_agg_block.reset((graph_input_struct.edge_node_agg_block).get());
+        edge_global_agg_block.reset((graph_input_struct.edge_global_agg_block).get());
+        node_global_agg_block.reset((graph_input_struct.node_global_agg_block).get());
+
+        num_nodes = graph_input_struct.num_nodes;
+        num_edges = graph_input_struct.edges.size();
+        num_node_features = graph_input_struct.num_node_features;
+        num_edge_features = graph_input_struct.num_edge_features;
+        num_global_features = graph_input_struct.num_global_features;
+        for(auto& it:graph_input_struct.edges){
+            receivers.emplace_back(it.first); 
+            senders.emplace_back(it.second);
+        }
+        fFileName = graph_input_struct.filename;
+        fName = fFileName.substr(0, fFileName.rfind("."));
+
+        std::time_t ttime = std::time(0);
+        std::tm* gmt_time = std::gmtime(&ttime);
+        fParseTime  = std::asctime(gmt_time);
+    }
+
+    void RModel_GNN::Generate(int batchSize){
+        std::string hgname;
+        GenerateHeaderInfo(hgname);
+
+        // Generating Infer function definition for Edge Update function
+        long next_pos;
+        fGC+="\n\nnamespace Edge_Update{\nstruct Session {\n";
+        std::vector<std::vector<std::size_t>> Update_Input = {{1, num_edge_features},{1, num_node_features},{1, num_node_features},{1, num_global_features}};
+        edges_update_block->Initialize();
+        edges_update_block->AddInputTensors(Update_Input);
+        fGC+=edges_update_block->GenerateModel(fName);
+        next_pos = edges_update_block->GetFunctionBlock()->WriteInitializedTensorsToFile(fName+".dat");
+        fGC+="};\n}\n";
+
+        fGC+="\n\nnamespace Node_Update{\nstruct Session {\n";
+        // Generating Infer function definition for Node Update function
+        Update_Input = {{1, num_edge_features},{1, num_node_features},{1, num_global_features}};
```

Comment:
I am getting this error here on macOS:
```
/Users/moneta/root/git/root/tmva/sofie/src/RModel_GNN.cxx:95:29: error: non-constant-expression cannot be narrowed from type 'int' to 'unsigned long' in initializer list [-Wc++11-narrowing]
        Update_Input = {{1, num_edge_features},{1, num_node_features},{1, num_global_features}};
                            ^~~~~~~~~~~~~~~~~
/Users/moneta/root/git/root/tmva/sofie/src/RModel_GNN.cxx:95:29: note: insert an explicit cast to silence this issue
        Update_Input = {{1, num_edge_features},{1, num_node_features},{1, num_global_features}};
        ```
        
Should maybe num_xxx_featres being also a size_t instead of an int ? or we should add an explicit cast. 
Similar error happening in other places (e.g. RModel_GraphINdipendent.cxx)

Reply:
casting issue, fixed that, thanks!